### ISSUE TITLE
IonPopup Problem

### ISSUE THREAD

**Issue by perusopersonale** [meteoric/meteor-ionic#132]
I don't know if this is an issue, maybe I'm just forcing it to do something that is not thought for.

I want to implement a global notification popup, so my code it's not in a template helper, but in the main code.

Client main.js:

Notifications.find().observe({
  added: function (notification) {
    IonPopup.show({
        title: 'notification',
        templateName: 'myTemplate', //this is not recognized
        buttons: [{
          text: 'Ok',
          type: 'button-positive',
          onTap: function() {
            Notifications.remove({_id:notification._id});
            IonPopup.close();
            }
        }]
      });
    }
There are two problems with this simple implementation:
1)if there is more than a notification so more than one popup, in console client(I'm testing with a browser) I get this error:
 Uncaught Error: Must be attached 
and the application crash.
2)templateName is not recognized

I'm not so good to understand if this is not possible with Meteoric or with meteor in general.
By now I'm simply display one popup(removing the other) and not using template and this is working.

**Comment by nickw** [comment]
@bastiRe do you have any thoughts on this?

@perusopersonale does `myTemplate` exist when you are calling `IonPopup.show()`?

**Comment by bastiRe** [comment]
@perusopersonale What exactly happens when you call your ionPopup with templateName? It works fine for me.

Concerning your first problem I also encountered this and I'll investigate if there's a workaround, but for now you can only have one popup at once.

**Comment by perusopersonale** [comment]
There was an error in mytemplate that's why it wasn't working, sorry...

**Comment by perusopersonale** [comment]
Ok, now I think I've understood what was happening.
my template is not a simple `HtmlRaw` but there is also a `Blaze.view`
so:
`Template[options.templateName].renderFunction().value` is returning `undefined`
because:
`Template[options.templateName].renderFunction()` is in my case an array, with a `Blaze.View` and `HtmlRaw`

**Comment by sircharleswatson** [comment]
@perusopersonale Thanks for asking this question! I could not figure out for the longest time what I was doing wrong. But I realized that it was because I was trying to do it similar to how the IonModal works.

**Comment by sircharleswatson** [comment]
@nickw or anyone else... Is there a way to pass a data object to the popup to show certain data in the template?

**Comment by nickw** [comment]
@sircharleswatson nope, doesn't look like that's currently possible, but maybe @bastiRe can confirm?

**Comment by sircharleswatson** [comment]
If not, I was actually just getting ready to try to implement the feature. I need it for an app I'm working on :)

I can add a PR when it's ready if that is something that would be welcome

**Comment by nickw** [comment]
@sircharleswatson absolutely!

**Comment by ericoe** [comment]
I'd like to do the following pattern: 

    confirm. 
        if yes, then do something. 
           alert to report if something bad happened

Here is the code:

			IonPopup.confirm({
				title: 'Title 1',
				template: 'Body 1',
				onOk: function()
				{
					// do something
					IonPopup.alert({title:'title 2', template:'body 2' });
				}
			});

This does not work... apparently shared state gets ripped up between calls. If I put in a "setTimeOut()" hack I can make the pattern work. Any advice on how to resolve the problem of showing 2 different popups in rapid time sequence?

**Comment by qdsang** [comment]
@nickw  @ericoe If you want to pop two, how to solve?

**Comment by ericoe** [comment]
roll your own... and on a related note, you might check: juliancwirko:s-alert

**Comment by Kiv** [comment]
@sircharleswatson Did you end up implementing a way to pass a data object to IonPopup? I also need this feature :)

**Comment by sircharleswatson** [comment]
@Kiv I got it mostly working but the code was pretty messy. I never got a chance to finish and clean it up unfortunately. and then i think I deleted the repository from my computer (and all local changes :/ ) Sorry!

**Comment by jstoup111** [comment]
Does anyone have some ideas on how to handle this? None of my event handlers are working.
